### PR TITLE
Refresh

### PR DESCRIPTION
**Pull request by erykpiast**
- [x] dependencies upgraded to the latest versions (requires dropping support for node v7)
- [x] old chrome runner replaced with puppeteer
- [x] all available options described in the program help
- [x] `--insecure` option added for testing local HTTPS servers
- [x] `--output` option added for controlling format of the output data (`pretty`, `table` and `csv` variants are available)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,14 +10,14 @@ function twoDigits(value) {
...
-            .map(measurement => {
-                return measurement
-                    .filter(timing => timing.name === name)
-                    .map(timing => timing.value);
-            }).map(t => t[0]).sort();
-    };
+    return (name) => measurements
+        .map(measurement =>
+            measurement
+                .filter(timing => timing.name === name)
```
Comment (adrianmroz) [inline index.js]: Use `find` here because you throw away rest of the values

File: chrome-runner.js [hunk 2]
```diff
@@ -1,53 +1,20 @@
...
-            ]
-        });
-
-        return launcher.run().then(() => launcher).catch(err => {
-            return launcher.kill().then(() => {
-                throw err;
-            }, console.error);
-        });
-    }
+    const results = await page.evaluate(() => {
```
Comment (adrianmroz) [inline chrome-runner.js]: extract and name that function

**Comment by msn0** [comment]
travis-ci (node 7) complains
```
const foundPath = await runMatcher({...options, cwd: directory});
		                    ^^^
SyntaxError: Unexpected token ...
```
Guys, I think we can safely drop support for node 7.

File: chrome-runner.js [hunk 3]
```diff
@@ -1,53 +1,20 @@
-const chrome = require('chrome-remote-interface');
-const { ChromeLauncher } = require('lighthouse/lighthouse-cli/chrome-launcher');
+const puppeteer = require('puppeteer');
 
-module.exports = async function (url) {
+module.exports = async function launch(url) {
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
```
Comment (plesiecki) [inline chrome-runner.js]: or `const [page] = await browser.pages();`

File: cli.js [hunk 4]
```diff
@@ -7,66 +7,90 @@ const teti = require('./');
...
-process.on('SIGINT', () => {
-    getHeadlessChromePid().then(p => {
-        fkill(p.pid).then(() => process.exit());
-    });
-});
+const runner = require(`./${cli.flags.runner}-runner`);
+const custom = Array.isArray(cli.flags.custom)
+    ? cli.flags.custom
+    : [cli.flags.custom];
+const { number, verbose, 'outputFormat': outputFormat } = cli.flags;
```
Comment (plesiecki) [inline cli.js]: just one `outputFormat` should be okay

File: cli.js [hunk 5]
```diff
@@ -77,18 +101,23 @@ teti({ url, num, notify, runner, custom }).then(output => {
...
+        chalk.yellow(o.metrics.median),
+        chalk.yellow(o.metrics.mean),
+        chalk.yellow(o.metrics.p95),
+        chalk.yellow(o.metrics.variance),
+        chalk.yellow(o.metrics.mad)
+    ]);
+
+    if (outputFormat === 'table') {
+        const t = table([headings, ...rows], {
+            align: Array.from(headings, () => 'r')
```
Comment (plesiecki) [inline cli.js]: why not `headings.map()`?

File: cli.js [hunk 6]
```diff
@@ -77,18 +101,23 @@ teti({ url, num, notify, runner, custom }).then(output => {
...
+        chalk.blue(o.name),
+        chalk.yellow(o.metrics.median),
+        chalk.yellow(o.metrics.mean),
+        chalk.yellow(o.metrics.p95),
+        chalk.yellow(o.metrics.variance),
+        chalk.yellow(o.metrics.mad)
+    ]);
+
+    if (outputFormat === 'table') {
+        const t = table([headings, ...rows], {
```
Comment (plesiecki) [inline cli.js]: have you considered `console.table`?

File: cli.js [hunk 7]
```diff
@@ -7,66 +7,90 @@ const teti = require('./');
...
+    flags: {
+        number: {
+            type: 'number',
+            alias: 'n',
+            default: 10
+        },
+        runner: {
+            type: 'string',
+            alias: 'r',
+            default: 'chrome'
```
Comment (bigsergey) [inline cli.js:36]: HeadlessChrome?

File: cli.js [hunk 8]
```diff
@@ -77,18 +101,23 @@ teti({ url, num, notify, runner, custom }).then(output => {
...
+
+    if (outputFormat === 'table') {
+        const t = table([headings, ...rows], {
+            align: Array.from(headings, () => 'r')
+        });
+
+        console.log(`\nResults for ${chalk.bgMagenta(url)} based on ${chalk.bgMagenta(number)} requests:\n`);
+        console.log(`${t}\n`);
+    } else {
+        console.log(csv([headings, ...rows]));
```
Comment (bigsergey) [inline cli.js]: console.table?

File: cli.js (hunk 7 above)
Comment (erykpiast) [reply cli.js:36]: In my opinion, there is no need for that. At the time we support only one runner - `chrome`. Testing "with a head" doesn't make much sense (unless debugging?) and the other provider that may be added eventually would be `firefox`. If we really need a switch for headless browsing, I would recommend orthogonal flag, like simply `--headless`

File: cli.js (hunk 8 above)
Comment (erykpiast) [reply cli.js]: That's a great idea! But I'd recommend to add it as an option, for two reasons:
 * it's not supported in each node.js version (only v10+ I believe)
 * it's not really equal to printing both "formatted table" and CSV - we have mandatory `(index)` column plus a lot of characters for borders that are meaningless but you have to copy them when copying the output
```
$ console.table(['foo', 'bar'], ['cat', 1], ['dog', 1])
┌─────────┬───┬─────┐
│ (index) │ 1 │ cat │
├─────────┼───┼─────┤
│    0    │   │     │
│    1    │   │     │
└─────────┴───┴─────┘
```

So, maybe: `pretty`, `table`, `csv`?

**Review by plesiecki — APPROVED** [review]

**Review by adrianmroz — APPROVED** [review]

**Comment by msn0** [comment]
Thanks Guys! Appreciate that + I'm asking for more :) :tada: 

📦  `teti@5.0.0`

**Pull request merged**